src/storage/data_table.cpp

```diff
@@ -20,7 +20,7 @@ void DataTable::Select(transaction::TransactionContext *txn,
                  "The projection never returns the version pointer, so it should have fewer attributes.");
```

Comment:
The following signature updates:

`void DataTable::Select(transaction::TransactionContext *const txn,
                       const TupleSlot slot,
                       ProjectedRow *const out_buffer) const`

`bool DataTable::Update(transaction::TransactionContext *const txn,
                       const TupleSlot slot,
                       const ProjectedRow &redo)`

`TupleSlot DataTable::Insert(transaction::TransactionContext *const txn,
                            const ProjectedRow &redo)`

`void DataTable::Rollback(const timestamp_t txn_id, const TupleSlot slot)`

`void DataTable::AtomicallyWriteVersionPtr(const TupleSlot slot,
                                          const TupleAccessStrategy &accessor,
                                          UndoRecord *const desired)`

`bool DataTable::HasConflict(UndoRecord *const version_ptr,
                            transaction::TransactionContext *const txn) const`